libcaf_core/caf/actor_system.cpp

```diff
@@ -23,6 +23,82 @@
 #include <unordered_map>
 #include <unordered_set>
 
+// -- printer actor ------------------------------------------------------------
+
+namespace caf::detail {
+
+struct printer_actor_state {
+  struct actor_data {
+    std::string current_line;
+    actor_data() {
+      // nop
+    }
+  };
+
+  using data_map = std::unordered_map<actor_id, actor_data>;
+
+  explicit printer_actor_state(event_based_actor* selfptr) : self(selfptr) {
+    // nop
+  }
+
+  event_based_actor* self;
+
+  data_map data;
+
+  actor_data* get_data(actor_id addr, bool insert_missing) {
+    if (addr == invalid_actor_id)
+      return nullptr;
+    auto i = data.find(addr);
+    if (i == data.end() && insert_missing)
+      i = data.emplace(addr, actor_data{}).first;
+    if (i != data.end())
+      return &(i->second);
+    return nullptr;
+  }
+
+  void flush(actor_data* what, bool forced) {
+    if (what == nullptr)
+      return;
+    auto& line = what->current_line;
+    if (line.empty() || (line.back() != '\n' && !forced))
+      return;
+    self->println(line);
+    line.clear();
+  }
+
+  behavior make_behavior() {
+    return {
+      [this](add_atom, actor_id aid, std::string& str) {
+        if (str.empty() || aid == invalid_actor_id)
+          return;
+        auto d = get_data(aid, true);
+        if (d != nullptr) {
+          d->current_line += str;
+          flush(d, false);
+        }
+      },
+      [this](flush_atom, actor_id aid) { flush(get_data(aid, false), true); },
+      [this](delete_atom, actor_id aid) {
+        auto data_ptr = get_data(aid, false);
+        if (data_ptr != nullptr) {
+          flush(data_ptr, true);
+          data.erase(aid);
+        }
+      },
+      [](redirect_atom, const std::string&, int) {
+        // nop
+      },
+      [](redirect_atom, actor_id, const std::string&, int) {
+        // nop
```

Comment:
Should the nop here stay, or is this a followup?

Reply:
That handler is in there for backwards compatibility only. We will deprecate `aout`, so this entire printer actor thingy will be removed eventually.